ProteoformSuiteInternal/ProteoformRelation.cs

```diff
@@ -57,19 +44,44 @@ public class ProteoformRelation : MassDifference
         public bool accepted { get; set; }
 
         public ProteoformRelation(Proteoform pf1, Proteoform pf2, ProteoformComparison relation_type, double delta_mass) 
-            : base(pf1, pf2, relation_type, delta_mass)
         {
-            if (Lollipop.neucode_labeled) this.lysine_count = pf1.lysine_count;
-            this.outside_no_mans_land = Math.Abs(delta_mass - Math.Truncate(delta_mass)) >= Lollipop.no_mans_land_upperBound ||
-                    Math.Abs(delta_mass - Math.Truncate(delta_mass)) <= Lollipop.no_mans_land_lowerBound;
+            this.connected_proteoforms[0] = pf1;
+            this.connected_proteoforms[1] = pf2;
+            this.relation_type = relation_type;
+            this.delta_mass = delta_mass;
+            instanceId = instanceCounter;
+            lock (Lollipop.proteoform_community) instanceCounter += 1; //Not thread safe
+
+            if (Lollipop.neucode_labeled)
+            {
+                this.lysine_count = pf1.lysine_count;
+            }
+
+            if (relation_type == ProteoformComparison.ExperimentalTheoretical || relation_type == ProteoformComparison.ExperimentalDecoy)
+            {
+                TheoreticalProteoform t = pf2 as TheoreticalProteoform;
+                double mass_tolerance = t.modified_mass / 1000000 * (double)Lollipop.mass_tolerance;
+                candidate_ptmset = t.generate_possible_added_ptmsets(nearestPTMs(delta_mass, relation_type).ToList(), delta_mass, mass_tolerance, Lollipop.all_mods_with_mass, t, t.sequence, Lollipop.rank_first_quartile)
+                    .OrderBy(x => (double)x.ptm_rank_sum + Math.Abs(Math.Abs(x.mass) - Math.Abs(delta_mass)) * 10E-6) // major score: delta rank; tie breaker: deltaM, where it's always less than 1
```

Comment:
My first thought when peak picking had been to use half the peak width to determine if a peak is close enough to a ptm/AA change. What's the reason to use mass tolerance instead? If that makes more sense I'll have the peak picking use that also!